From d6b179a5addef6456325adf241eb8dcaf8e4c3d6 Mon Sep 17 00:00:00 2001
From: Thomas Hellstrom <thellstrom@vmware.com>
Date: Thu, 9 Jan 2014 13:28:22 +0100
Subject: [PATCH 17/19] vmware: Fix build errors and warnings

A previous commit and the hosted merge unfortunately brought in some build
errors / warnings on early X servers.

Signed-off-by: Thomas Hellstrom <thellstrom@vmware.com>
Reviewed-by: Jakob Bornecrantz <jakob@vmware.com>
---
 src/vmware_bootstrap.c    | 1 +
 src/vmwarevideo.c         | 4 ++--
 vmwgfx/vmwgfx_driver.c    | 2 +-
 vmwgfx/vmwgfx_overlay.c   | 4 ++--
 vmwgfx/vmwgfx_tex_video.c | 2 +-
 5 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/src/vmware_bootstrap.c b/src/vmware_bootstrap.c
index 57f8ae9..ed6c740 100644
--- a/src/vmware_bootstrap.c
+++ b/src/vmware_bootstrap.c
@@ -34,6 +34,7 @@
 #include "xf86Pci.h"		/* pci */
 #include "vm_device_version.h"
 #include "vmware_bootstrap.h"
+#include <stdint.h>
 
 #if GET_ABI_MAJOR(ABI_VIDEODRV_VERSION) < 6
 #include "xf86Resources.h"
diff --git a/src/vmwarevideo.c b/src/vmwarevideo.c
index 8d7d171..745c71f 100644
--- a/src/vmwarevideo.c
+++ b/src/vmwarevideo.c
@@ -82,7 +82,7 @@
 #define VMWARE_VID_MAX_HEIGHT   2048
 
 #define VMWARE_VID_NUM_ENCODINGS 1
-static const XF86VideoEncodingRec vmwareVideoEncodings[] =
+static XF86VideoEncodingRec vmwareVideoEncodings[] =
 {
     {
        0,
@@ -108,7 +108,7 @@ static XF86ImageRec vmwareVideoImages[] =
 };
 
 #define VMWARE_VID_NUM_ATTRIBUTES 2
-static const XF86AttributeRec vmwareVideoAttributes[] =
+static XF86AttributeRec vmwareVideoAttributes[] =
 {
     {
         XvGettable | XvSettable,
diff --git a/vmwgfx/vmwgfx_driver.c b/vmwgfx/vmwgfx_driver.c
index dc05b86..2d38d2a 100644
--- a/vmwgfx/vmwgfx_driver.c
+++ b/vmwgfx/vmwgfx_driver.c
@@ -130,7 +130,7 @@ vmwgfx_hookup(ScrnInfoPtr pScrn)
 }
 
 void
-vmwgfx_modify_flags(CARD32 *flags)
+vmwgfx_modify_flags(uint32_t *flags)
 {
     *flags &= ~(HW_IO);
     vmwgfx_hosted_modify_flags(flags);
diff --git a/vmwgfx/vmwgfx_overlay.c b/vmwgfx/vmwgfx_overlay.c
index c8c6bb9..2c0d7fa 100644
--- a/vmwgfx/vmwgfx_overlay.c
+++ b/vmwgfx/vmwgfx_overlay.c
@@ -85,7 +85,7 @@ typedef uint8_t uint8;
 #define VMWARE_VID_MAX_HEIGHT   2048
 
 #define VMWARE_VID_NUM_ENCODINGS 1
-static const XF86VideoEncodingRec vmwareVideoEncodings[] =
+static XF86VideoEncodingRec vmwareVideoEncodings[] =
 {
     {
        0,
@@ -111,7 +111,7 @@ static XF86ImageRec vmwareVideoImages[] =
 };
 
 #define VMWARE_VID_NUM_ATTRIBUTES 2
-static const XF86AttributeRec vmwareVideoAttributes[] =
+static XF86AttributeRec vmwareVideoAttributes[] =
 {
     {
         XvGettable | XvSettable,
diff --git a/vmwgfx/vmwgfx_tex_video.c b/vmwgfx/vmwgfx_tex_video.c
index 9fd8f22..0803a99 100644
--- a/vmwgfx/vmwgfx_tex_video.c
+++ b/vmwgfx/vmwgfx_tex_video.c
@@ -82,7 +82,7 @@ static XF86VideoFormatRec Formats[NUM_FORMATS] = {
    {15, TrueColor}, {16, TrueColor}, {24, TrueColor}
 };
 
-static const XF86VideoEncodingRec DummyEncoding[1] = {
+static XF86VideoEncodingRec DummyEncoding[1] = {
    {
       0,
       "XV_IMAGE",
-- 
1.8.5.3

